thousandeyes/resource_agent_to_server.go

```diff
@@ -0,0 +1,297 @@
+			// 	Elem: &schema.Resource{
+			// 		Schema: map[string]*schema.Schema{
+			// 			"monitor_id": {
+			// 				Type:        schema.TypeInt,
+			// 				Description: "monitor id",
+			// 				Optional:    true,
+			// 			},
+			// 		},
+			// 	},
+			// },
+			"agents": {
+				Type:        schema.TypeList,
+				Description: "agents to use ",
+				Required:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"agent_id": {
+							Type:        schema.TypeInt,
+							Description: "agent id",
+							Optional:    true,
+						},
+					},
+				},
+			},
+		},
+		Create: resourceAgentServerCreate,
+		Read:   resourceAgentServerRead,
+		Update: resourceAgentServerUpdate,
+		Delete: resourceAgentServerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+func resourceAgentServerRead(d *schema.ResourceData, m interface{}) error {
+	client := m.(*thousandeyes.Client)
+
+	log.Printf("[INFO] Reading Thousandeyes Agent %s", d.Id())
+	id, _ := strconv.Atoi(d.Id())
+	agent, err := client.GetAgentServer(id)
+	if err != nil {
+		return err
+	}
+
+	d.Set("name", agent.TestName)
+	d.Set("id", agent.TestId)
+	d.Set("interval", agent.Interval)
+	d.Set("description", agent.Description)
+	d.Set("bandwidth_measurements", agent.BandwidthMeasurements)
+	d.Set("bgp_measurements", agent.BgpMeasurements)
+	d.Set("mtu_measurements", agent.MtuMeasurements)
+	d.Set("num_path_traces", agent.NumPathTraces)
+	d.Set("port", agent.Port)
+	d.Set("protocol", agent.Protocol)
+	d.Set("server", agent.Server)
+	// d.Set("bgp_monitors", agent.BgpMonitors)
+	d.Set("agents", agent.Agents)
+
+	return nil
+}
+
+func resourceAgentServerUpdate(d *schema.ResourceData, m interface{}) error {
+	client := m.(*thousandeyes.Client)
+
+	log.Printf("[INFO] Updating ThousandEyes Test %s", d.Id())
+	d.Partial(true)
+	id, _ := strconv.Atoi(d.Id())
+	var update thousandeyes.AgentServer
+
+	if d.HasChange("name") {
+		update.TestName = d.Get("name").(string)
+	}
+
+	if d.HasChange("description") {
+		update.Description = d.Get("description").(string)
+	}
+
+	if d.HasChange("bandwidth_measurements") {
+		update.BandwidthMeasurements = d.Get("bandwidth_measurements").(int)
+	}
+	if d.HasChange("protocol") {
+		update.Protocol = d.Get("protocol").(string)
+	}
+	if d.HasChange("interval") {
+		update.Interval = d.Get("interval").(int)
+	}
+	if d.HasChange("mtu_measurements") {
+		update.MtuMeasurements = d.Get("mtu_measurements").(int)
+	}
+	if d.HasChange("num_path_traces") {
+		update.NumPathTraces = d.Get("num_path_traces").(int)
+	}
+
+	if d.HasChange("port") {
+		update.Port = d.Get("port").(int)
+	}
+
+	if d.HasChange("probe_mode") {
+		update.ProbeMode = d.Get("probe_mode").(string)
+	}
+
+	if d.HasChange("bgp_measurements") {
+		update.BgpMeasurements = d.Get("bgp_measurements").(int)
+	}
+	if d.HasChange("agents") {
+		update.Agents = expandAgents(d.Get("agents").([]interface{}))
+	}
+
+	// Need to figure this out
```

Comment:
can we create an issue to track to track this, just so it doesnt get lost and forgotten about.

Reply:
https://github.com/william20111/terraform-provider-thousandeyes/issues/7